proto/pulumi/provider.proto

```diff
@@ -215,32 +220,105 @@ message GetSchemaResponse {
     string schema = 1; // the JSON-encoded schema.
 }
 
+// `ConfigureRequest` is the type of requests sent as part of a [](pulumirpc.ResourceProvider.Configure) call. Requests
+// include both provider-specific inputs (`variables` or `args`) and provider-agnostic ("protocol") configuration
+// (`acceptSecrets`, `acceptResources`, and so on).
 message ConfigureRequest {
-    // the input properties for the provider, with nested values JSON-encoded; please read from `args` instead.
+    // :::{warning}
+    // `variables` is deprecated; `args` should be used instead wherever possible.
+    // :::
+    //
+    // A map of input properties for the provider. Compound values, such as nested objects, should be JSON encoded so
+    // that they too can be passed as strings. For instance, the following configuration:
+    //
+    // ```
+    // {
+    //   "a": 42,
+    //   "b": {
+    //     "c": "hello",
+    //     "d": true
+    //   }
+    // }
+    // ```
+    //
+    // should be encoded as:
```

Comment:
Not obvious how secrets are encoded. Will unknowns ever be passed here? I think Configure might not be ever called in preview, but if that's so it is great to document.

Reply:
Tracked in #17691